.github/workflows/build-magma-linux.yml

```diff
@@ -0,0 +1,39 @@
+name: Build magma linux
+
+on:
+  push:
+    branches:
+      main
+    paths:
+      - magma/*
+      - common/install_conda.sh
+      - .github/workflows/build-magma-linux.yml
+  pull_request:
+    paths:
+      - magma/*
+      - common/install_conda.sh
+      - .github/workflows/build-magma-linux.yml
+
+jobs:
+  build-linux-magma:
+    runs-on: linux.2xlarge
+    strategy:
+      matrix:
+        cuda_version: ["115"]
+    steps:
+      - name: Checkout PyTorch builder
+        uses: actions/checkout@v2
+      - name: Build Magma Cuda
+        working-directory: magma
+        run: |
+          make magma-cuda${{ matrix.cuda_version }}
+      - name: Save as artifact
+        uses: actions/upload-artifact@v2
+        with:
+          path: magma/output/linux-64/magma-cuda*.bz2
+      - name: Push MAGMA to anaconda
```

Comment:
We should probably limit the push to just on pushes to `main`

Reply:
Yes added if: ${{ github.event_name == 'push' }}